Expand implementation instructions (#35)
w3c/vc-di-ecdsa-test-suite: README.md

To add your implementation to this test suite, you will need to add 2 endpoints
to your implementation manifest.
- A credential issuer endpoint (`/credentials/issue`) in the `issuers` property.

>   - An optional id property specifying the issuer can be set alongside the endpoint.
>   - If provided, the specified issuer id will be used as the issuer property on verifiable credentials
>     in the tests.
>   - The issuer id should be easy for other implementers to dereference.
>     - It is recommended the issuer id be `did:key`.

Reviewer: ```suggestion
  - An optional `id` property may be set alongside the `endpoint`.
  - If provided, the specified `issuer.id` will be added to Verifiable Credentials
    in the tests.
  - If present, the `issuer.id` MUST use the `did:key` method and MUST dereferenceable.
```

Reviewer: Simplifying and clarifying. Pretty sure the `id` values MUST be dereferenceable--and that the test suite only currently supports `did:key` making both those things requirements, not recommendations.

- A credential verifier endpoint (`/credentials/verify`) in the `verifiers`
property.
- An optional `vcHolder` endpoint can be added for `ecdsa-sd-2023` selective disclosure tests.
  - The vcHolder endpoint route is meant to be `/credentials/derive`
  - The endpoint is expected to accept a JSON object with 2 properties:
    - `options` which is a JSON object.
    - `options.selectivePointers` which is an array of strings.
    - `verifiableCredential` which is a previously signed JSON verifiable credential.
    - The endpoint needs to use `ecdsa-sd-2023` for the derived verifiable credential.

All endpoints will require a cryptosuite tag of `ecdsa-rdfc-2019`,
`ecdsa-jcs-2019`, and/or `ecdsa-sd-2023`. Alongside this cryptosuite tag, you
will need to specify the `supportedEcdsaKeyTypes` property, parallel to `tags`
listing the ECDSA key types issuable or verifiable by your implementation.
Currently, the test suite supports `P-256` and `P-384` ECDSA key types.

> A `vcHolder` tag is required for the `vcHolder` endpoints.

Reviewer: Can we avoid this tag if they've supplied the additional options data and tagged the endpoint with `ecdsa-sd-2023`?

To add your implementation to this test suite, you will need to add 2 endpoints
to your implementation manifest.
- A credential issuer endpoint (`/credentials/issue`) in the `issuers` property.

>   - The issuer id will be used as the issuer property on VCs in the tests.

Reviewer 2: ```suggestion
  - The issuer id will be used as the issuer property on verifiable credentials
    in the tests.
```

- A credential verifier endpoint (`/credentials/verify`) in the `verifiers`
property.

> - An optional `vcHolder` endpoint maybe added for `ecdsa-sd-2023` selective disclosure tests.
>   - The vcHolder endpoint route should be `/credentials/derive`
>   - The endpoint needs to accept a JSON object with 2 properties:
>     - `options` which is a JSON object.
>     - `options.selectivePointers` which is an array of strings.
>     - `verifiableCredential` which is a previously signed JSON VC.
>     - The endpoint needs to use `ecdsa-sd-2023` for the derived VC.

Reviewer 2: ```suggestion
- An optional `vcHolder` endpoint can be added for `ecdsa-sd-2023` selective disclosure tests.
  - The vcHolder endpoint route is meant to be `/credentials/derive`
  - The endpoint is expected to accept a JSON object with 2 properties:
    - `options` which is a JSON object.
    - `options.selectivePointers` which is an array of strings.
    - `verifiableCredential` which is a previously signed JSON verifiable credential.
    - The endpoint needs to use `ecdsa-sd-2023` for the derived verifiable credential.
```

All endpoints will require a cryptosuite tag of `ecdsa-rdfc-2019`,
`ecdsa-jcs-2019`, and/or `ecdsa-sd-2023`. Alongside this cryptosuite tag, you

> must also specify the `supportedEcdsaKeyTypes` property, parallel to `tags`

Reviewer 2: ```suggestion
will need to specify the `supportedEcdsaKeyTypes` property, parallel to `tags`
```

  "name": "My Company",
  "implementation": "My implementation",
  "issuers": [{
    "id": "did:key:myIssuer1#keyFragment",
    "endpoint": "https://mycompany.example/credentials/issue",
    "method": "POST",
    "supportedEcdsaKeyTypes": ["P-256", "P-384"]
    "tags": ["ecdsa-rdfc-2019"]
  }, {
    "id": "did:key:myIssuer#issuer2",
    "endpoint": "https://mycompany.example/credentials/issue",
    "method": "POST",
    "supportedEcdsaKeyTypes": ["P-256"]
    "tags": ["ecdsa-jcs-2019"]
  }, {
    "id": "did:key:myIssuer3",
    "endpoint": "https://mycompany.example/credentials/issue",
    "method": "POST",
    "supportedEcdsaKeyTypes": ["P-256"]
    "tags": ["ecdsa-sd-2023"]
  }],
  "verifiers": [{
    "endpoint": "https://mycompany.example/credentials/verify",
    "method": "POST",
    "supportedEcdsaKeyTypes": ["P-256", "P-384"]
    "tags": [
      "ecdsa-rdfc-2019", "ecdsa-jcs-2019", "ecdsa-sd-2023"
    ]
  }],
  "vcHolders": [{
    "id": "did:key:myIssuer1#keyFragment",
    "endpoint": "https://mycompany.example/credentials/derive",

>     "tags": ["vcHolder"]

Reviewer: Should this metadata be associated with only the `ecdsa-sd-2023` issuer rather than appear here on it's own?

Author: This design is not that best and I think this could be accomplished with a compound tag like `tags: ['ecdsa-sd-2023', 'derive']` and then just list it in the issuers section, but this PR is about documentation and also once the local issuer is in, we can possibly drop this tag entirely. So maybe an issue about vcHolder and the issues around it would be better? maybe all it needs is the `ecdsa-sd-2023` in addition to the `vcHolder` in the manifest?
